caddyhttp/staticfiles/fileserver.go

```diff
@@ -40,6 +40,12 @@ func (fs FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, err
 	return fs.serveFile(w, r, r.URL.Path)
 }
 
+// calculateEtag produces a strong etag by default. Prefix the result with "W/" to convert this into a weak one.
+// see https://tools.ietf.org/html/rfc7232#section-2.3
+func calculateEtag(d os.FileInfo) string {
+	return fmt.Sprintf(`"%x-%x"`, d.ModTime().Unix(), d.Size())
```

Comment:
This is a good question.  Generally, I think you're right to be more concerned about the inefficiency of scanning the whole file to compute a hash. And the use of timestamp+length is *not* a complete guarantee that the file contents haven't changed.

But the bullish high-performance argument is that timestamp+length is good enough in all normal use-cases. This is what most people will be happy with, I think, and is what Nginx does. After all, strong etags lead to more aggressive caching, boosting perceived performance significantly.

Thinking about this a little more, the same argument suggests that the compressed artefacts should also have strong etags (based on timestamp+length).

If you think Caddy should be more accommodating, perhaps we can provide an option to force the etags to be weak, or to disable them completely..

And there's more work to be done on 304s.

So I am now thinking that maybe this needs a little more work to

a) use strong etags for all served resources by default
b) provide an option to use weak etags instead when required.
c) implement 304-not-modified checks on If-None-Match conditional requests.

I'll take another look tonight.

[also of interest: PR191 https://github.com/mholt/caddy/issues/191]